x/curating/spec/README.md

```diff
@@ -0,0 +1,209 @@
+# Curating Module Specification
+
+## Abstract
+
+This module provides functionality to curate social media content and earn rewards in cryptocurrency via market-based mechanisms.
+
+## Concepts
+
+First, a post is created with a `post` transaction, taking a hash of the content, and a deposit. The deposit is returned after a curation window.
+
+Posts are curated with `upvote` transactions and `moderate` transactions.
+
+Upvotes are based on quadratic voting, and take number of votes as an argument. They also takes a deposit that is returned after the curation window. Deposits are mean't to keep the system honest. For example, governance may choose to slash the deposit of upvotes that are engaging in collusion or a Sybil attack.
+
+Moderate transactions are for signalling that a post contains spam or illegal content. Moderated posts immediately go to governance for adjudication. Curation is paused during this time. For posts deemed as inappropriate, upvoters get slashed and the moderator gets a reward. For posts not deemed as inappropriate, curation resumes while the moderator gets slashed.
+
+Only data for posts currently being curated are stored on-chain. Everything else is deleted after the curation window completes and rewards are paid out.
+
+## State
+
+### Parameters
+
+```go
+type Params struct {
+	CurationWindow       	 time.Duration
+	DefaultPostDeposit       sdk.Coin
+	DefaultUpvoteDepost  	 sdk.Coin
+	DefaultModerationDeposit sdk.Coin
+	DefaultVoteAmount        sdk.Coin // amount for 1 vote
+	MaxNumVotes              int // upper-limit on QV voting
+	RewardPoolAllocation 	 sdk.Dec // from inflation
+}
+```
+
+### Base Types
+
+```go
+type Post struct {
+	VendorID        uint64
```

Comment:
Are a list of vendors kept anywhere yet? Or is that future work? Guess, `int64` is fine for now if it's the later... especially as we'll only be supporting Twitter initially.

Reply:
Might be worth adding a super simple vendor struct that we can evolve over time?

Reply:
Good idea. The intention with `vendor_id` is for anyone to permissionlessly use the protocol for any social service. So I don't want to hardcode it in an enum or struct. But what we could do is only allow a `vendor_id` in a certain range that's set by a parameter. Then people could petition governance to add another vendor like Discord, and the parameter will increment for the new vendor. I'll make it a parameter.